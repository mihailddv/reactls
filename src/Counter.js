import React, { useState } from "react";

export const Counter = () => {
  const [count, setCount] = useState(0);
  const increment = () => {
    setCount(count => count + 1);
  };

  const decrement = () => {
    setCount(count => count - 1);
  };

  return (
    <>
      <div className="t-text">{count}</div>
      <button className="t-btn-increment" onClick={increment}>Increment</button>
      <button className="t-btn-descrement" onClick={decrement}>Decrement</button>
    </>
  );
};
