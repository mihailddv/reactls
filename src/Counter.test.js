// Компонент Counter - простой счётчик
// Напишите для него тесты.
// Убедитесь, что вы протестировали всю функциональнось.
// Также проверьте что компонент рендерится верно, используя Snapshot тест.

// * Задание со звёздочкой - выполнять не обязательно

// Вынесите логику в хук и протестируйте его

import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { Counter } from './Counter';

describe('Counter', () => {

    describe('Component render correctly', () => {

        it('renders correctly', () => {
            const tree = renderer.create(<Counter />).toJSON();
            expect(tree).toMatchSnapshot();
        });

    });



    describe('Counter correctly', () => {

        const container = shallow(<Counter />);
        const incrementButton = container.find('.t-btn-increment');
        const descrementButton = container.find('.t-btn-descrement');

        it('increment correctly', () => {
            expect(container.find('.t-text').text()).toEqual('0');
            incrementButton.simulate('click');
            expect(container.find('.t-text').text()).toEqual('1');

            incrementButton.simulate('click');
            expect(container.find('.t-text').text()).toEqual('2');
        });

        it('descrement correctly', () => {
            expect(container.find('.t-text').text()).toEqual('2');
            descrementButton.simulate('click');
            expect(container.find('.t-text').text()).toEqual('1');

            descrementButton.simulate('click');
            expect(container.find('.t-text').text()).toEqual('0');
        });

    });

});