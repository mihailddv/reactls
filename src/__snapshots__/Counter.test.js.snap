// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component render correctly renders correctly 1`] = `
Array [
  <div
    className="t-text"
  >
    0
  </div>,
  <button
    className="t-btn-increment"
    onClick={[Function]}
  >
    Increment
  </button>,
  <button
    className="t-btn-descrement"
    onClick={[Function]}
  >
    Decrement
  </button>,
]
`;
